internal/server/device/device_utils_disk.go

```diff
@@ -344,15 +345,43 @@ func DiskVMVirtiofsdStart(execPath string, inst instance.Instance, socketPath st
 	}
 
 	// Start the virtiofsd process in non-daemon mode.
-	args := []string{"--fd=3", fmt.Sprintf("--cache=%s", cacheOption), "-o", fmt.Sprintf("source=%s", sharePath)}
-	proc, err := subprocess.NewProcess(cmd, args, logPath, logPath)
-	if err != nil {
-		return nil, nil, err
-	}
+	args := []string{"--fd=3", fmt.Sprintf("--cache=%s", cacheOption), fmt.Sprintf("--shared-dir=%s", sharePath)}
 
 	if len(idmaps) > 0 {
 		idmapSet := &idmap.Set{Entries: idmaps}
-		proc.SetUserns(idmapSet.ToUIDMappings(), idmapSet.ToGIDMappings())
+		sort.Sort(idmapSet)
+
+		var lastUID int64
+		var lastGID int64
+
+		for _, entry := range idmapSet.Entries {
+			if entry.IsUID {
+				args = append(args, fmt.Sprintf("--translate-uid=map:%d:%d:%d", entry.NSID, entry.HostID, entry.MapRange))
+
+				args = append(args, fmt.Sprintf("--translate-uid=forbid-guest:%d:%d", lastUID, entry.NSID-1))
+				lastUID = entry.NSID
+			}
+
+			if entry.IsGID {
+				args = append(args, fmt.Sprintf("--translate-gid=map:%d:%d:%d", entry.NSID, entry.HostID, entry.MapRange))
+
+				args = append(args, fmt.Sprintf("--translate-gid=forbid-guest:%d:%d", lastGID, entry.NSID-1))
+				lastGID = entry.NSID
+			}
+		}
+
+		if lastUID < 4294967295 {
```

Comment:
Hm, but if there was only a uid, and no gid, then
lastGid will still be its default value and therefore
pass the < 4294967295 test, and you'll add an entry
for that.  Is that what you wanted?

Reply:
If no gid is passed (very bad idea), we should ban all GIDs, so I'd expect a 0-4294967295 reject range to be put in place.

Reply:
But that actually brings in a good point that I should initialize at -1 as with current logic, we'd end up only forbidding 1-4294967295, allowing the most dangerous one...

Reply:
For extra fun: https://gitlab.com/virtio-fs/virtiofsd/-/issues/198